README.md

```diff
@@ -1,223 +1,350 @@
+    * A pool that wants to begin supporting Sv2 but is not yet ready to completely overhaul their
+      existing Sv1 server can use these libraries to construct a Sv2<->Sv1 proxy server. This proxy
+      server lives on the upstream pool end under the pool's control, between the downstream
+      devices on the miner's side (either a miner's proxy or the mining devices themselves), and
+      the most upstream Sv1 pool server. This most upstream Sv1 pool server is likely the already
+      existing pool server logic. In this way, pools can begin supporting Sv2 without waiting to
+      redo their entire server infrastructure to be Sv2 compatible only. This Sv2<->Sv1 pool proxy
+      will only allow the pool to support Standard channels (not Group or Extended), therefore this
+      should be used as a temporary measure before completely upgrading to a Sv2-only pool.
+    * A Sv2- or Sv1-compatible pool that wants to offer job negotiation capabilities to their
+      customers (miner's mining to their pool) can use this library to self-host a Job Negotiator
+      and Template Provider (a minimum amount of modification is still required).
+* Make this primitives available also to non Rust user.
+
+## 2. Structure
+```
+stratum
+  └─ examples
+  │   └─ interop-cpp
+  │   └─ interop-cpp-no-cargo
+  │   └─ ping-pong-with-noise
+  │   └─ ping-pong-without-noise
+  │   └─ sv1-client-and-server
+  │   └─ sv2-proxy
+  │   └─ template-provider-test
+  └─ experimental
+  │   └─ coinbase-negotiator
+  └─ protocols
+  │   └─ v1
+  │   └─ v2
+  │       └─ binary-sv2
+  │       │   └─ binary-sv2
+  │       │   └─ no-serde-sv2
+  │       │   └─ serde-sv2
+  │       └─ codec-sv2
+  │       └─ const-sv2
+  │       └─ framing-sv2
+  │       └─ noise-sv2
+  │       └─ roles-logic-sv2
+  │       └─ subprotocols
+  │       │   └─ common-messages
+  │       │   └─ job-negotiation
+  │       │   └─ mining
+  │       │   └─ template-distribution
+  │       └─ sv2-ffi
+  └─ roles/v2
+  │   └─ mining-proxy
+  │   └─ pool
+  │   └─ test-utils
+  │   │   └─ mining-device
+  │   │   └─ pool
+  └─ test
+  └─ utils
+      └─ buffer
+      └─ network-helpers
+```
+
+This workspace is divided in 6 macro-areas:
+1. `protocols`: Core Stratum V2 and V1 libraries.
+1. `roles`: The Sv2 roles as binaries to be executed, including the mining-device, mining-proxy,
+            and pool. This gives good examples of the actual implementation/use of the available
+            libraries.
+1. `utils`: Offers an alternative buffer to use that is more efficient, but less safe than the
+            default buffers used. Also has network helpers.
+1. `examples`: Several example implementations of various use cases.
+1. `test`: Integration tests.
+1. `experimental`: Experimental logic that is not yet specified in the protocol or in a protocol
+                   extension.
+
+All dependencies related to the testing and benchmarking of this repository is optional and is
```

Comment:
All dependencies related to the testing and benchmarking in these workspace's crates are optional, and are included in the binary only when we test or we execute a benchmark. For that reason we do not include them under **External dependencies** in the below lists.